pkg/catalogv2/system/system.go

```diff
@@ -120,7 +120,10 @@ func (m *Manager) onTrigger(_ string, obj *catalog.ClusterRepo) (*catalog.Cluste
 		return obj, nil
 	}
 
-	m.trigger <- struct{}{}
+	select {
+	case m.trigger <- struct{}{}:
+	default:
+	}
```

Comment:
Do we want to error or re-enqueue in the `default` case to ensure this eventually gets reconciled?

Reply:
I don't think so because if this happens the queue already has one item in it.  That chan is of length 1.

Reply:
Hmm, thats an interesting point. If the catalog is rancher-charts and its blocked, then default executes and nothing happens until it specifically gets changed/refreshed again.